worker/upload.go

```diff
@@ -292,14 +303,24 @@ func (w *worker) uploadMultiPart(ctx context.Context, r io.Reader, bucket, path,
 		return "", fmt.Errorf("couldn't add multi part: %w", err)
 	}
 
-	// if packing was enabled try uploading packed slabs in a separate goroutine
+	// if packing was enabled try uploading packed slabs
 	if up.packing {
-		go w.tryUploadPackedSlabs(up.rs, up.contractSet)
+		// if we reached the buffer size limit, we upload packed slabs synchronously
+		if bufferSizeLimitReached {
+			ctx, cancel := context.WithTimeout(ctx, defaultPackedSlabsUploadTimeout)
+			defer cancel()
+			_, err := w.uploadPackedSlabs(ctx, defaultPackedSlabsUploadTimeout, up.rs, up.contractSet, blockedUploadPackedSlabsLimit)
```

Comment:
wanted your input on these values here, so I think if we upload synchronously to drop below the buffer soft limit we should probably do a bit more than 2, but we don't want to stall this request for too long either...

I figured adding a 10' timeout and thus also increasing the lock duration to 10' to try and upload 5 packed slabs could be possible solution? It depends on the redundancy settings of course so we could take those into account too? We could also just use a limit of `2` like we used to.. bit arbitrary here and lots of tradeoffs

Reply:
I probably wouldn't overcomplicate this with multiple slabs and a timeout. If we hit the limit, we upload 1 slab synchronously. That way we don't push beyond the limit with this upload. Afterwards we still trigger the background thread on this worker if it isn't running already. That will then upload more slabs if necessary.

So essentially if you run this PR on a 1TB buffer, all your small uploads will block for 1 full slab being uploaded + 1 thread will continuously upload in the background until the buffer is at 0 complete slabs.